### PR TITLE
maintenance: enhance test case for parser and chore schema

### PR DESCRIPTION
**Pull request by MH4GF**
Details are described in the in-line comments.

### CODE REVIEW

File: frontend/packages/db-structure/src/parser/sql/postgresql/converter.ts [hunk 1]
```diff
@@ -118,10 +122,6 @@ export const convertToDBStructure = (ast: RawStmtWrapper[]): DBStructure => {
               ?.filter(isConstraintNode)
               .some((c) => c.Constraint.contype === 'CONSTR_PRIMARY') ||
             false,
-          increment:
-            colDef.typeName?.names
-              ?.filter(isStringNode)
-              .some((n) => n.String.sval === 'serial') || false,
```
Comment (MH4GF) [inline frontend/packages/db-structure/src/parser/sql/postgresql/converter.ts:124]: I've remove unnecessary increment field.

File: frontend/packages/db-structure/src/parser/__tests__/testcase.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        comment: override?.comment ?? null,
+      }),
+    },
+  })
+
+export const parserTestCases = {
+  'table comment': userTable({
+    comment: 'store our users.',
+  }),
+}
```
Comment (MH4GF) [inline frontend/packages/db-structure/src/parser/__tests__/testcase.ts:35]: To unify the behaviour across multiple parsers, an object defining common test cases has been added.

File: frontend/packages/db-structure/src/parser/sql/postgresql/converter.ts [hunk 3]
```diff
@@ -108,7 +108,11 @@ export const convertToDBStructure = (ast: RawStmtWrapper[]): DBStructure => {
           unique:
             colDef.constraints
               ?.filter(isConstraintNode)
-              .some((c) => c.Constraint.contype === 'CONSTR_UNIQUE') || false,
+              .some((c) =>
+                ['CONSTR_UNIQUE', 'CONSTR_PRIMARY'].includes(
+                  c.Constraint.contype ?? '',
+                ),
+              ) || false,
```
Comment (MH4GF) [inline frontend/packages/db-structure/src/parser/sql/postgresql/converter.ts:115]: Behaviour aligned with schemarb-parser; 
`unique: true` for primary key.

File: frontend/packages/db-structure/src/parser/sql/postgresql/index.test.ts [hunk 4]
```diff
@@ -36,23 +37,18 @@ describe(processor, () => {
...
-
-      expect(result).toEqual(expected)
+      expect(result).toEqual(parserTestCases['table comment'])
     })
 
     it('not null', async () => {
       const result = await processor(/* sql */ `
         CREATE TABLE users (
-          id SERIAL PRIMARY KEY,
+          id BIGSERIAL PRIMARY KEY,
```
Comment (MH4GF) [inline frontend/packages/db-structure/src/parser/sql/postgresql/index.test.ts:51]: Behaviour aligned with schemarb-parser: test data of type id is `BIGSERIAL`.

File: frontend/packages/db-structure/src/parser/__tests__/testcase.ts (hunk 2 above)
Comment (hoshinotsuyoshi) [reply frontend/packages/db-structure/src/parser/__tests__/testcase.ts:35]: > If this policy is acceptable, other test cases will be changed.

ask:
`userTable` will be defined only here(`src/parser/__tests__/testcase.ts`) ?

File: frontend/packages/db-structure/src/parser/__tests__/testcase.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        indices: {
+          ...override?.indices,
+        },
+        comment: override?.comment ?? null,
+      }),
+    },
+  })
+
+export const parserTestCases = {
+  'table comment': userTable({
```
Comment (hoshinotsuyoshi) [inline frontend/packages/db-structure/src/parser/__tests__/testcase.ts:32]: Nice 👍 
It seems great that type inference works!

File: frontend/packages/db-structure/src/parser/__tests__/testcase.ts (hunk 2 above)
Comment (MH4GF) [reply frontend/packages/db-structure/src/parser/__tests__/testcase.ts:35]: @hoshinotsuyoshi 

> userTable will be defined only here(src/parser/__tests__/testcase.ts) ?

Yes, I think it's better that way.

**Review by hoshinotsuyoshi — APPROVED** [review]
LGTM, but I’d like to hear others' opinions as well!

File: frontend/packages/db-structure/src/parser/__tests__/testcase.ts (hunk 2 above)
Comment (sasamuku) [reply frontend/packages/db-structure/src/parser/__tests__/testcase.ts:35]: @MH4GF 
Looks nice 👍
If this policy is accepted, we have to unify type name as well, right?
I would like to know how to change type name for Oracle, SQL Server and so on.

Comment (MH4GF) [reply frontend/packages/db-structure/src/parser/__tests__/testcase.ts:35]: @sasamuku Yes, there will inevitably be subtle differences.
I will consider how to override this as we move forward with implementation. 👍🏻

**Review by sasamuku — APPROVED** [review]
Thank you for enhancement!

**Pull request merged**